File: cc/002.cc
// Project Euler Problem 002
//
// By considering the terms in the Fibonacci sequence whose values do
// not exceed four million, find the sum of the even-valued terms.

#include "fib.h"
#include <cstdio>

int main(int argc, const char* argv[]) {
  Fib<int> fib;
  int total = 0;
  int f = fib.next();
  while (f < 4e6) {
    if (! (f&1)) {
      total += f;
    }
    f = fib.next();
  }
  printf("%d\n", total);
}

File: go/problems/002.go
// Project Euler Problem 002
//
// By considering the terms in the Fibonacci sequence whose values do
// not exceed four million, find the sum of the even-valued terms.

package problems

import "fmt"

func Problem002() string {
	sum := 0
	for f1, f2 := 1, 1; f1 <= 4000000; f1, f2 = f1+f2, f1 {
		if f1%2 == 0 {
			sum += f1
		}
	}
	return fmt.Sprintf("%d", sum)
}
